clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/agent/KubernetesV2CachingAgent.java

```diff
@@ -139,7 +140,7 @@ protected CacheResult buildCacheResult(Map<KubernetesKind, List<KubernetesManife
         .map(rs -> {
           try {
             CacheData cacheData = KubernetesCacheDataConverter.convertAsResource(accountName, rs, relationships.get(rs));
-            if(credentials.getOnlySpinnakerManaged() && cacheData.getAttributes().get("application").equals("")) {
+            if (credentials.getOnlySpinnakerManaged() && !StringUtils.isNotEmpty((String) cacheData.getAttributes().get("application"))) {
```

Comment:
Whoops, I gave bad advice! I would just use `StringUtils.isEmpty` instead of negating `isNotEmpty`

Reply:
Ha, it's no problem. I should have checked the docs. Ill fix that real quick.